File: src/_088/kotlin/Solution.kt
package _088.kotlin

import java.util.*

/**
 * @author relish
 * @since 2018/04/24
 */
class Solution {
    fun merge(nums1: IntArray, m: Int, nums2: IntArray, n: Int): Unit {
        var index = n + m - 1
        var i = m - 1
        var j = n - 1

        while ((i >= 0 || j >= 0) && index >= 0) {
            if (j < 0 && i in 0..(m - 1)) {
                nums1[index--] = nums1[i--]
                continue
            }
            if (i < 0 && j in 0..(n - 1)) {
                nums1[index--] = nums2[j--]
                continue
            }
            if (nums1[i] > nums2[j]) {
                nums1[index--] = nums1[i--]
            } else {
                nums1[index--] = nums2[j--]
            }
        }
    }
}

fun main(args: Array<String>) {
    val arr1 = intArrayOf(1, 2, 3, 0, 0, 0)
    val arr2 = intArrayOf(2, 5, 6)
    Solution().merge(arr1, 3, arr2, 3)
    println(Arrays.toString(arr1))
}

File: src/_088/Solution.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    nums1.splice(m)
    nums2.splice(n)
    nums1.push(...nums2)
    for(var i = 0; i < nums1.length - 1; i++) {
        for(var j = 0; j < nums1.length - 1 - i; j++) {
            if(nums1[j] > nums1[j+1]) {
                var x = nums1[j]
                nums1[j] = nums1[j+1]
                nums1[j+1] = x
            }
        }
    }
};